dev/PushNotifications/PushNotificationTelemetry.h

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "..\WindowsAppSDK_Insights\WindowsAppRuntimeInsights.h"
+#include <wrl\wrappers\corewrappers.h>
+
+DECLARE_TRACELOGGING_CLASS(PushNotificationTelemetryProvider, "Microsoft.WindowsAppSDK.Notifications.PushNotificationTelemetry", (0x7c1b07ef, 0xa7c0, 0x56d1, 0x54, 0x56, 0x38, 0x5e, 0xbd, 0x44, 0x12, 0xb2));
+// {7c1b07ef-a7c0-56d1-5456-385ebd4412b2}
+
+class PushNotificationTelemetry : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TELEMETRY_CLASS(PushNotificationTelemetry, PushNotificationTelemetryProvider);
+
+public:
+    DEFINE_EVENT_METHOD(ChannelRequestedByApi)(
+        winrt::hresult hr,
+        bool isAppPackaged,
+        const winrt::guid& remoteId) noexcept try
+        {
+            if (c_maxEventLimit >= UpdateLogEventCount())
+            {
+                TraceLoggingClassWriteMeasure(
+                    "ChannelRequestedByApi",
+                    TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                    _GENERIC_PARTB_FIELDS_ENABLED,
+                    TraceLoggingHexUInt32(hr, "OperationResult"),
+                    TraceLoggingBool(isAppPackaged, "IsAppPackaged"),
+                    TraceLoggingWideString(to_hstring(remoteId).data(), "RemoteId"));
```

Comment:
Can this be replaced by DEFINE_COMPLIANT_TRACELOGGING_EVENT_PARAM3(ChannelRequestedByApi, int, operationResult, bool, isAppPackaged, PCWSTR, remoteId); ?

and 

PushNotificationTelemetry::ChannelRequestedByApi(hr, isAppPackaged, to_hstring(remoteId).data());


This is so that you don't have to reference _GENERIC_PARTB_FIELDS_ENABLED in your code.

Reply:
It almost can but the solution you're proposing will fail to compile because it re-use the same eventId for DEFINE_COMPLIANT_TRACELOGGING_EVENT_PARAM3 as was used for DEFINE_EVENT_METHOD. The function signatures will be too similar and the compiler will get all confused.

We can go around this by introducing a new name for the eventId but this somewath obfuscates the code, i.e.: the name you'll see in the logs, won't be the same that the provider exposes in its API.

Reply:
What is wrong with refering to _GENERIC_PARTB_FIELDS_ENABLED in the code?

Reply:
_GENERIC_PARTB_FIELDS_ENABLED is an internal implementation detail. It should not bleed into any other components. Ideally, it should be hidden similar to PartA. We are not able to hide it because the telemetry is MACRO based.

Reply:
I don't think it's a reasonable expectation. _GENERIC_PARTB_FIELDS_ENABLED is very public and you won;t be able to police its usage for ever. People will see it and use it when they need it.

Have you thought about introducing a new TraceLoggingClassWriteMeasure or whatever TraceLoggingWrite function it wraps that would have applied _GENERIC_PARTB_FIELDS_ENABLED. If people have access to the correct construct they need in their code, that would certainly help deter people from _GENERIC_PARTB_FIELDS_ENABLED.

Reply:
For example, if there was a version of TraceLoggingClassWriteMeasure that applied _GENERIC_PARTB_FIELDS_ENABLED, I'd be very happy to use it.